lib/spack/docs/build_settings.rst

```diff
@@ -209,6 +209,192 @@ Specific limitations include:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once all dependencies are satisfied or if there are no dependencies, then the
+prefix determined by Spack may not be correct in the presence of a nonstandard
+directory structure. This is rarely the case because Spack contains
+package-specific code to deal with these quirks. Before manually setting the
+prefix in the ``packages.yaml`` file in addition to a list of modules, it is
+strongly suggested to check the package name and the module list again. 
+
+The files of OpenMPI on CentOS 7 use a nonstandard directory structure. For
+example on x86-64, the libraries are in ``/usr/lib64/openmpi`` (on x86-64
+machines) instead of ``/usr/lib64`` or ``/usr/lib``. Consider the ``module show
+mpi`` output:
+
+.. code-block:: console
+
+    [john@c7 ~]# module show mpi
+    -------------------------------------------------------------------
+    /etc/modulefiles/mpi/openmpi-x86_64:
+
+    conflict	 mpi
+    prepend-path	 PATH /usr/lib64/openmpi/bin
+    prepend-path	 LD_LIBRARY_PATH /usr/lib64/openmpi/lib
+    prepend-path	 PYTHONPATH /usr/lib64/python2.7/site-packages/openmpi
+    prepend-path	 MANPATH /usr/share/man/openmpi-x86_64
+    [snip]
+
+Given this module output, ``spack external find`` will determine
+``/usr/lib64/openmpi`` as the prefix when it is actually ``/usr``. This
+incorrect prefix can cause the error message below when building packages::
+
+    ==> Error: AttributeError: Query of package 'openmpi' for 'headers' failed
+    	prefix : None
+    	spec : openmpi@1.10.7%gcc@4.8.5~cuda+cxx_exceptions fabrics=none ~java~legacylaunchers~memchecker~pmi schedulers=none ~sqlite3~thread_multiple+vt arch=linux-centos7-x86_64
+    	queried as : openmpi
+    	extra parameters : []
+
+The solution in this case is to manually edit the prefix in the
+``packages.yaml`` file (here for CentOS 7):
+
+.. code-block:: yaml
+
+    packages:
+      openmpi:
+        externals:
+        - spec: openmpi@1.10.7%gcc@4.8.5~cuda+cxx~cxx_exceptions~java~memchecker~pmi~sqlite3~static~thread_multiple~wrapper-rpath
+          prefix: /usr
+          modules: [mpi]
+
+~~~~~~~~~~~
+Metamodules
+~~~~~~~~~~~
+
+Consider the following ``module show`` output:
+
+.. code-block:: console
+
+    $ module show intel-all
+    -------------------------------------------------------------------
+    /gpfslocalsup/pub/module-rh/modulefiles/intel-all/2019.4:
+
+    conflict        intel-all
+    module          load intel-compilers/19.0.4
+    module          load intel-mkl/19.0.4
+    module          load intel-mpi/19.0.4
+    module          load intel-vtune/19.0.4
+    module          load intel-advisor/19.0.4
+    module          load intel-tbb/19.0.4
+    module          load intel-itac/19.0.4
+    -------------------------------------------------------------------
+
+The module ``intel-all`` is obviously a metamodule because it only loads other
+modules and its ``module show`` output cannot be used to derive the needed
```

Comment:
I would rephrase:

> and its ``module show`` output cannot be used to derive the needed environment changes to use

as

> ; the module contents do not include direct manipulations of `PATH`, which Spack depends on to determine the package prefix.

Also regarding: 

> For this reason, metamodules *cannot* be used in ``packages.yaml`` files.

Could a metamodule be used *in combination* with a lower level module? e.g. if the metamodule loads needed dependencies it might be faster than the example in the section "Missing Modules Dependencies"

I might rephrase that line as

> For this reason, metamodules cannot generally be used in place of the direct package module when defining external packages.

Reply:
fixed in c766605dcf825b9062b426799e22bf63cb384808